File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def solve(self, root: Optional[TreeNode], ans: int) -> int:
        if not root:
            return 0
        ans = ans * 10 + root.val
        if not root.left and not root.right:
            return ans
        return self.solve(root.left, ans) + self.solve(root.right, ans)
        
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        return self.solve(root, 0)

File: LeetCode/0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int solve(TreeNode root, int ans) {
        if (root == null) {
            return 0;
        }
        ans = ans * 10 + root.val;
        if (root.left == null && root.right == null) {
            return ans;
        }
        return solve(root.left, ans) + solve(root.right, ans);
    }
    public int sumNumbers(TreeNode root) {
        return solve(root, 0);
    }
}